### PR TITLE
BUG: Fix Cubic Spline Read Only issues

### PR DESCRIPTION
**Pull request by dmadeka**
Fixes issue #9740

### CODE REVIEW

**Comment by dmadeka** [comment]
@tylerjereddy Not sure what the pypy3 issues are - the tests that are failing dont seem related to the change made

**Comment by tylerjereddy** [comment]
Try rebasing on latest master after the resolution from #9763.

File: scipy/interpolate/_ppoly.pyx [hunk 1]
```diff
@@ -40,7 +40,7 @@ DEF MAX_DIMS = 64
 @cython.cdivision(True)
 def evaluate(double_or_complex[:,:,::1] c,
              double[::1] x,
```
Comment (ev-br) [inline scipy/interpolate/_ppoly.pyx]: While we're at it, might also allow for `const double x`

**Review by ev-br — COMMENTED** [review]
If you're feeling ambitious, can also allow for readonly arrays of breakpoints `x`, in evaluate and `PPoly.integrate`.
Possibly also `BPoly.from_derivatives`.
However, all this is optional and can be done in a follow-up PR. Other than that, LGTM.

File: scipy/interpolate/_ppoly.pyx (hunk 1 above)
Comment (dmadeka) [reply scipy/interpolate/_ppoly.pyx]: Done! Tests pass - Ill take on `PPoly.integrate` and `BPoly.from_derivatives` in another PR?

Comment (dmadeka) [reply scipy/interpolate/_ppoly.pyx]: Im not sure why _differential_evolution is failing - it doesnt seem to have an obvious connection to `PPoly` and doesnt fail locally

**Comment by dmadeka** [comment]
Looks like a timeout error on `test_parallel`. Maybe rerunning the workflow will pass the test?

**Comment by ev-br** [comment]
Thanks @dmadeka , keep them coming!

**Comment by dmadeka** [comment]
@ev-br PPoly should be easy. But looks like the `BPoly` updates will have to wait till the fused types issue is solved

https://github.com/scipy/scipy/blob/master/scipy/interpolate/_ppoly.pyx#L1070

**Comment by dmadeka** [comment]
@ev-br did this make 1.2.1? Doesnt look like it --

**Pull request merged**